src/main/java/seedu/address/logic/parser/AddPersonalTaskCommandParser.java

```diff
@@ -0,0 +1,133 @@
+package seedu.address.logic.parser;
+
+import seedu.address.logic.commands.AddPersonalTaskCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+import seedu.address.model.Schedule;
+import seedu.address.model.person.exceptions.DurationParseException;
+import seedu.address.model.person.exceptions.TimingClashException;
+import seedu.address.model.personal.PersonalTask;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.ResolverStyle;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+/**
+ * Parses input arguments and creates a new AddPersonalTaskCommand object.
+ */
+public class AddPersonalTaskCommandParser {
+
+    private static final String MESSAGE_TASK_TIMING_CLASHES = "This task clashes with another task";
+    private static final String MESSAGE_INVALID_DATE_TIME = "The input date and time is invalid";
+    private static final String MESSAGE_INVALID_DURATION = "The duration format is invalid";
+    private static final int LENGTH_OF_DATE_TIME = 16;
+    private static final int INDEX_OF_START_OF_DURATION = LENGTH_OF_DATE_TIME + 1;
+    private static final int INVALID_INDEX = -1;
+    private static final String NO_DESCRIPTION = "";
+    private static final String DURATION_VALIDATION_REGEX = "([0-9]|1[0-9]|2[0-3])\\b(h)\\b([0-5][0-9])\\b(h)\\b";
+
+    private static String duration;
+    private static String description;
+    private static DateTimeFormatter formatter = DateTimeFormatter
+            .ofPattern("yyyy/mm/dd HH:mm")
```

Comment:
should the format be changed to "dd/mm/yyyy HH:mm"?

Reply:
can also